packages/lab/src/typography/Text.tsx

```diff
@@ -72,97 +71,99 @@ export interface TextProps extends HTMLAttributes<HTMLElement> {
   /**
    * Match styling to a specified heading
    */
-  styleAs?: "h1" | "h2" | "h3" | "h4";
+  styleAs?: "h1" | "h2" | "h3" | "h4" | "label";
 }
 
-export const Text = forwardRef<HTMLElement, TextProps>(function Text(
-  props,
-  ref
-) {
-  const {
-    children,
-    className,
-    elementType = "div",
-    expanded,
-    marginBottom,
-    marginTop,
-    maxRows,
-    onOverflowChange,
-    showTooltip,
-    style,
-    styleAs,
-    tooltipProps,
-    tooltipText,
-    truncate = false,
-    ...restProps
-  } = props;
+export type TextProps<E extends ElementType> = TextPropsBase<E> &
```

Comment:
Given `elementType` defaults to `div`, i wonder whether setting the default on Generic here as well